front/lib/api/assistant/agent_usage.ts

```diff
@@ -22,261 +13,134 @@ import { redisClient } from "@app/lib/redis";
+  limit?: number;
+}): Promise<agentUsage[]> {
+  const owner = await Workspace.findOne({ where: { sId: workspaceId } });
   if (!owner) {
     throw new Error(`Workspace ${workspaceId} not found`);
   }
-  const { agentMessageCountKey, agentUserCountKey } = _getKeys({
-    agentConfigurationId,
-    workspaceId,
-  });
 
-  const existCount = await redis.exists([
-    agentMessageCountKey,
-    agentUserCountKey,
-  ]);
-  if (existCount === 0) {
-    // Sorted sets for this agent usage do not exist, we'll populate them
-    // by fetching the data from the database.
-    // We need to ensure that only one process is going through the populate code path
-    // so we are using redis.incr() to act as a non blocking lock.
-    const populateLockKey = `agent_usage_populate_${workspaceId}_${agentConfigurationId}`;
-    const needToPopulate = (await redis.incr(populateLockKey)) === 1;
+  const { agentMessageCountKey, agentUserCountKey } =
+    _getUsageKeys(workspaceId);
+  const redis = providedRedis ?? (await redisClient());
 
-    // Keeping the lock key around for 10 minutes, which essentially gives 10 minutes
-    // to create the sorted sets, before running the risk of a race conditions.
-    // A race condition in creating the sorted sets would result in double counting
-    // usage of the agent.
-    const populateTimeoutSec = 60 * 10; // 10 minutes
-    await redis.expire(populateLockKey, populateTimeoutSec);
-    if (!needToPopulate) {
-      return;
-    }
+  try {
+    const agentMessageCountTTL = await redis.ttl(agentMessageCountKey);
 
-    // We are safe to populate the sorted sets until the Redis populateLockKey expires.
-    // Get all mentions for this agent that have a messageId smaller than messageId
-    // and that happened within the last 30 days.
-    const mentions = await Mention.findAll({
-      where: {
-        ...{
-          agentConfigurationId: agentConfigurationId,
-          createdAt: {
-            [Op.gt]: literal(`NOW() - INTERVAL '30 days'`),
-          },
-        },
-        ...(messageModelId ? { messageId: { [Op.lt]: messageModelId } } : {}),
-      },
-      include: [
-        {
-          model: Message,
-          required: true,
-          include: [
-            {
-              model: UserMessage,
-              as: "userMessage",
-              required: true,
-            },
-            {
-              model: Conversation,
-              as: "conversation",
-              required: true,
-              where: {
-                workspaceId: owner.id,
-              },
-            },
-          ],
-        },
-      ],
-    });
-    for (const mention of mentions) {
-      // No need to promise.all() here, as one Redis connection can only execute one command
-      // at a time.
-      if (mention.message?.userMessage) {
-        await signalInRedis({
-          agentConfigurationId,
+    // agent mention count doesn't exist
+    if (agentMessageCountTTL === -2) {
```

Comment:
is there a constant provided by the redis client that's more readable?

Reply:
Not that I am aware of unfortunately